src/Tools/dotnet-monitor/Egress/AzureBlob/AzureBlobEgressProvider.cs

```diff
@@ -116,8 +124,23 @@ public AzureBlobEgressProvider(ILogger<AzureBlobEgressProvider> logger)
                 // Write blob headers
                 await blobClient.SetHttpHeadersAsync(CreateHttpHeaders(artifactSettings), cancellationToken: token);
 
-                // Write blob metadata
-                await blobClient.SetMetadataAsync(artifactSettings.Metadata, cancellationToken: token);
+                try
+                {
+                    Dictionary<string, string> mergedMetadata = artifactSettings.Metadata.Union(artifactSettings.CustomMetadata).ToDictionary(pair => pair.Key, pair => pair.Value);
+
+                    // Write blob metadata
+                    await blobClient.SetMetadataAsync(mergedMetadata, cancellationToken: token);
+                }
+                catch (Exception ex) when (ex is InvalidOperationException || ex is RequestFailedException)
```

Comment:
If you take any of my suggestions above, then remove this catcher.

Reply:
Why are we removing this catcher? I was using this to handle invalid names (e.g. `&MyKey`), which is a separate issue from having duplicate keys

Reply:
So, I am already doing this - in the catch statements, I add in _just_ the built-in metadata, since the issue is with the custom metadata. I originally tried to apply them both separately, but I think that was overwriting whatever I applied first with what I applied second (as opposed to merging them - I could be mistaken though). That being said, I can adjust the try/catch so that it only is wrapped around `SetMetadataAsync`